src/features/bivariate_color_manager/components/SentimentsCombinationsList/SentimentsCombinationsList.tsx

```diff
@@ -70,28 +91,91 @@ const SentimentsCombinationsList = memo(
-              <td className={clsx(s.label)}>{verticalLabel}</td>
-              <td className={clsx(s.label)}>{horizontalLabel}</td>
-            </tr>
-          ))}
+          {rows.map(({ key, legend, maps, verticalLabel, horizontalLabel }) => {
+            const rowSelected = selectedRowKeys[key];
+            const selectRow = () =>
+              setSelectedRowKeys({
+                ...selectedRowKeys,
+                [key]: !selectedRowKeys[key],
+              });
+
+            const { vertical, horizontal } = data[key];
+            const verticalList = Object.values(vertical).sort(
+              sortDescendingByQuality,
+            );
+            const horizontalList = Object.values(horizontal).sort(
+              sortDescendingByQuality,
+            );
+
+            return (
+              <React.Fragment key={key}>
+                <tr
+                  onClick={selectRow}
+                  className={clsx(rowSelected && s.rowSeleted)}
+                >
+                  <td>
+                    <div className={clsx(s.legendWrapper)}>
+                      {legend && (
+                        <MiniLegend
+                          legend={invertClusters(legend.steps, 'label')}
+                        />
+                      )}
+                    </div>
+                  </td>
+                  <td className={clsx(s.centered)}>{maps}</td>
+                  <td className={clsx(s.label)}>{verticalLabel}</td>
+                  <td className={clsx(s.label)}>{horizontalLabel}</td>
+                </tr>
+
+                <CSSTransitionWrapper
+                  in={rowSelected}
+                  timeout={300}
+                  unmountOnExit
+                  appear
+                  classNames={fadeClassNames}
+                >
+                  <tr className={clsx(s.sublist)}>
+                    <td />
+                    <td />
+                    <td>
+                      {verticalList.map(({ label }) => (
+                        <div className={clsx(s.sublistRow)} key={label}>
+                          {label}
+                        </div>
+                      ))}
+                    </td>
+                    <td>
+                      {horizontalList.map(({ label }) => (
+                        <div className={clsx(s.sublistRow)} key={label}>
+                          {label}
+                        </div>
+                      ))}
+                    </td>
+                  </tr>
+                </CSSTransitionWrapper>
+              </React.Fragment>
+            );
+          })}
         </tbody>
       </table>
     );
   },
 );
 
+const CSSTransitionWrapper = ({ children, ...props }: CSSTransitionProps) => {
+  const nodeRef = useRef(null);
+
+  return (
+    <CSSTransition {...props} nodeRef={nodeRef}>
+      <>
+        {Children.map(children, (child) =>
+          isValidElement(child) ? cloneElement(child, { ref: nodeRef }) : child,
```

Comment:
Be careful with `cloneElement`, is expensive operation, but here it clone only one children - tr, so in should be okay

Reply:
do you know a better way of passing some prop to children? 
maybe render function pattern would work better?

```
<CSSTransitionWrapper
                  in={rowSelected}
                  timeout={300}
                  unmountOnExit
                  appear
                  classNames={fadeClassNames}
                >
                (ref) => 
                  <tr ref={ref} className={s.sublist}>
```